dist_run.py

```diff
@@ -499,12 +529,17 @@ def decode_in_flight(token):
     # output formatted response via last pp group and tp rank 0
     if pp_rank == last_pp_rank and tp_rank == 0:
         # `res` is a list of tensors, each being a batch of generated token ids
-        res = torch.stack(res, dim=1)
-        res_list = res.tolist()
-        response = tokenizer.decode(res_list)
-        for i in range(len(response)):
-            logger.info(f"Prompt: {color.green}{prompt[i]} {color.reset}")
-            logger.info(f"Response: {color.red}{response[i]} {color.reset}")
+
+        res_stacked = torch.stack(res, dim=1)
+        res_list = res_stacked.tolist()
+
+        # Decode the output as comprehension instead of loop
+        responses = [tokenizer.decode(sequence) for sequence in res_list]
```

Comment:
Hmm, did the previous code not work in case of variable length? Just curious.
`response = tokenizer.decode(res_list)`